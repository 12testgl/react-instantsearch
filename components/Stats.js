var React = require('react');

var Template = require('./Template');

class Stats extends React.Component {
  render() {
    var template = this.props.template;
    var defaultTemplate = this.props.defaultTemplate;
    var templatesConfig = this.props.templatesConfig;
    var transformData = this.props.transformData;
    var data = {
      hasManyResults: this.props.nbHits > 1,
      hasNoResults: this.props.nbHits === 0,
      hasOneResult: this.props.nbHits === 1,
      hitsPerPage: this.props.hitsPerPage,
      nbHits: this.props.nbHits,
      nbPages: this.props.nbPages,
      page: this.props.page,
      processingTimeMS: this.props.processingTimeMS,
      query: this.props.query
    };

    return (
      <Template
        data={data}
        transformData={transformData}
        template={template}
        defaultTemplate={defaultTemplate}
        config={templatesConfig}
      />
    );
  }
}

Stats.propTypes = {
  hitsPerPage: React.PropTypes.number,
  nbHits: React.PropTypes.number,
  nbPages: React.PropTypes.number,
  page: React.PropTypes.number,
  processingTimeMS: React.PropTypes.number,
  template: React.PropTypes.oneOfType([
    React.PropTypes.func,
    React.PropTypes.string
  ]),
  defaultTemplate: React.PropTypes.oneOfType([
    React.PropTypes.func,
    React.PropTypes.string
  ]).isRequired,
  transformData: React.PropTypes.func,
  templatesConfig: React.PropTypes.object.isRequired,
  query: React.PropTypes.string
};

module.exports = Stats;
