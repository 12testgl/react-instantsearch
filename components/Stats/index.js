var React = require('react');

var Template = require('../Template/');

class Stats extends React.Component {
  render() {
    var template = this.props.template;
    var data = {
      nbHits: this.props.nbHits,
      hasNoResults: this.props.nbHits === 0,
      hasOneResult: this.props.nbHits === 1,
      hasManyResults: this.props.nbHits > 1,
      processingTimeMS: this.props.processingTimeMS
    };

    return (
      <Template
        data={data}
        template={template}
      />
    );
  }
}

Stats.propTypes = {
  nbHits: React.PropTypes.number,
  processingTimeMS: React.PropTypes.number,
  template: React.PropTypes.oneOfType([
    React.PropTypes.func,
    React.PropTypes.string
  ]).isRequired
};

module.exports = Stats;
