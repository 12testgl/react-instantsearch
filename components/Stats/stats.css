.ais-stats {
}
.ais-stats--time {
  font-size: small;
}
