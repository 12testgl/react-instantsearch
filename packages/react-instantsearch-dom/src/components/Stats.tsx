import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';
import { translatable } from 'react-instantsearch-core';
import { createClassNames } from '../core/utils';

const cx = createClassNames('Stats');

export type StatsTranslationKeyProps = {
  isSmartSorted?: boolean;
  n: number;
  nSorted?: number;
  ms: number;
};

export type StatsComponent = {
  className?: string;
  isSmartSorted: boolean;
  nbHits: number;
  nbSortedHits?: number;
  processingTimeMS: number;
  translate(key: string, props: StatsTranslationKeyProps): string;
};

const Stats: React.FC<StatsComponent> = ({
  className = '',
  isSmartSorted,
  nbHits,
  nbSortedHits,
  processingTimeMS,
  translate,
}) => {
  return (
    <div className={classNames(cx(''), className)}>
      <span className={cx('text')}>
        {translate('stats', {
          n: nbHits,
          nSorted: nbSortedHits,
          isSmartSorted,
          ms: processingTimeMS,
        })}
      </span>
    </div>
  );
};

Stats.propTypes = {
  className: PropTypes.string,
  isSmartSorted: PropTypes.bool.isRequired,
  nbHits: PropTypes.number.isRequired,
  nbSortedHits: PropTypes.number,
  processingTimeMS: PropTypes.number.isRequired,
  translate: PropTypes.func.isRequired,
};

export default translatable({
  stats: ({
    n,
    nSorted,
    isSmartSorted = false,
    ms,
  }: StatsTranslationKeyProps): string =>
    isSmartSorted && nSorted !== undefined && n !== nSorted
      ? `${nSorted.toLocaleString()} relevant results sorted out of ${n.toLocaleString()} in ${ms.toLocaleString()}ms`
      : `${n.toLocaleString()} results found in ${ms.toLocaleString()}ms`,
})(Stats);
