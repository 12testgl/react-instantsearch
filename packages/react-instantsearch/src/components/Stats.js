import PropTypes from 'prop-types';
import React, { Component } from 'react';

import BaseWidget from './BaseWidget';
import translatable from '../core/translatable';
import { classNamesNew } from './classNames.js';

const widgetClassName = 'Stats';
const cx = classNamesNew(widgetClassName);

class Stats extends Component {
  static propTypes = {
    translate: PropTypes.func.isRequired,
    nbHits: PropTypes.number.isRequired,
    processingTimeMS: PropTypes.number.isRequired,
  };

  render() {
    const { translate, nbHits, processingTimeMS } = this.props;
    return (
      <BaseWidget widgetClassName={widgetClassName}>
        <span {...cx(['text'])}>
          {translate('stats', nbHits, processingTimeMS)}
        </span>
      </BaseWidget>
    );
  }
}

export default translatable({
  stats: (n, ms) =>
    `${n.toLocaleString()} results found in ${ms.toLocaleString()}ms`,
})(Stats);
