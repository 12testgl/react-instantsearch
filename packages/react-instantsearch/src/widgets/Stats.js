import PropTypes from 'prop-types';
import React from 'react';
import BaseWidget from './BaseWidget';
import connectStats from '../connectors/connectStats.js';
import StatsComponent from '../components/Stats.js';
import classNames from '../components/classNames';

const cx = classNames('Stats');

/**
 * The Stats component displays the total number of matching hits and the time it took to get them (time spent in the Algolia server).
 * @name Stats
 * @kind widget
 * @propType {node} [header] - Adds a header to the widget.
 * @propType {node} [footer] - Adds a footer to the widget.
 * @themeKey ais-Stats - the root div of the widget
 * @themeKey ais-Stats-header - the header of the widget (optional)
 * @themeKey ais-Stats-body - the body of the widget
 * @themeKey ais-Stats-text - the text of the widget - the count of items for each item
 * @themeKey ais-Stats-footer - the footer of the widget (optional)
 * @translationkey stats - The string displayed by the stats widget. You get function(n, ms) and you need to return a string. n is a number of hits retrieved, ms is a processed time.
 * @example
 * import React from 'react';
 *
 * import { Stats, InstantSearch } from 'react-instantsearch/dom';
 *
 * export default function App() {
 *   return (
 *     <InstantSearch
 *       appId="latency"
 *       apiKey="6be0576ff61c053d5f9a3225e2a90f76"
 *       indexName="ikea"
 *     >
 *       <Stats />
 *     </InstantSearch>
 *   );
 * }
 */

const Widget = props => (
  <BaseWidget cx={cx} header={props.header} footer={props.footer}>
    <StatsComponent cx={cx} {...props} />
  </BaseWidget>
);

Widget.propTypes = {
  header: PropTypes.node,
  footer: PropTypes.node,
};

export default connectStats(Widget);
