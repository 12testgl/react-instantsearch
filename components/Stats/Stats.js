var React = require('react');

var Template = require('../Template');
var bem = require('../../lib/utils').bemHelper('ais-stats');
var cx = require('classnames');

require('style?prepend!raw!./stats.css');

class Stats extends React.Component {
  render() {
    var data = {
      hasManyResults: this.props.nbHits > 1,
      hasNoResults: this.props.nbHits === 0,
      hasOneResult: this.props.nbHits === 1,
      hitsPerPage: this.props.hitsPerPage,
      nbHits: this.props.nbHits,
      nbPages: this.props.nbPages,
      page: this.props.page,
      processingTimeMS: this.props.processingTimeMS,
      query: this.props.query,
      cssClasses: {
        root: cx(bem(null), this.props.cssClasses.root),
        time: cx(bem('time'), this.props.cssClasses.time)
      }
    };

    return (
      <this.props.Template data={data} templateKey="body" />
    );
  }
}

Stats.propTypes = {
  hitsPerPage: React.PropTypes.number,
  nbHits: React.PropTypes.number,
  nbPages: React.PropTypes.number,
  page: React.PropTypes.number,
  processingTimeMS: React.PropTypes.number,
  query: React.PropTypes.string
};

module.exports = Stats;
