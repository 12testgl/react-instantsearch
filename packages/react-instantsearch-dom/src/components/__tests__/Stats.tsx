import React from 'react';
import renderer from 'react-test-renderer';
import Stats from '../Stats';

describe('Stats', () => {
  it('renders with default props', () => {
    const tree = renderer.create(
      <Stats nbHits={42} isSmartSorted={false} processingTimeMS={0} />
    );

    expect(tree.toJSON()).toMatchSnapshot();
  });

  it('accepts a custom className', () => {
    const tree = renderer.create(
      <Stats
        isSmartSorted={false}
        className="MyCustomStats"
        nbHits={42}
        processingTimeMS={0}
      />
    );

    expect(tree.toJSON()).toMatchSnapshot();
  });

  it('renders rely on isSmartSorted', () => {
    const tree = renderer.create(
      <Stats
        isSmartSorted={true}
        nbHits={42}
        nbSortedHits={21}
        processingTimeMS={0}
      />
    );

    expect(tree.toJSON()).toMatchSnapshot();
  });

  it('renders default implementation if nbHits is equal to nbSortedHits', () => {
    const tree = renderer.create(
      <Stats
        isSmartSorted={true}
        nbHits={42}
        nbSortedHits={42}
        processingTimeMS={0}
      />
    );

    expect(tree.toJSON()).toMatchSnapshot();
  });
});
