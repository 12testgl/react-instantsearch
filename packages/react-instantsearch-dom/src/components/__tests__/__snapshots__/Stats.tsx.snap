// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Stats accepts a custom className 1`] = `
<div
  className="ais-Stats MyCustomStats"
>
  <span
    className="ais-Stats-text"
  >
    42 results found in 0ms
  </span>
</div>
`;

exports[`Stats renders default implementation if nbHits is equal to nbSortedHits 1`] = `
<div
  className="ais-Stats"
>
  <span
    className="ais-Stats-text"
  >
    42 results found in 0ms
  </span>
</div>
`;

exports[`Stats renders rely on isSmartSorted 1`] = `
<div
  className="ais-Stats"
>
  <span
    className="ais-Stats-text"
  >
    21 relevant results sorted out of 42 in 0ms
  </span>
</div>
`;

exports[`Stats renders with default props 1`] = `
<div
  className="ais-Stats"
>
  <span
    className="ais-Stats-text"
  >
    42 results found in 0ms
  </span>
</div>
`;
